### PR TITLE
Migrate GetBlock to MigratingEndpointAdapter

### LINKED ISSUE #5466: Migrate GetBlock V1 to MigratingEndpointAdapter

### PR DESCRIPTION
**Pull request by courtneyeh**
fixes #5466 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlock.java [hunk 1]
```diff
@@ -25,37 +29,56 @@
...
+            .operationId("getBlock")
+            .summary("Get block")
+            .description(
+                "Retrieves block details for given block id.\n\n"
+                    + "__NOTE__: only phase 0 blocks are returned, use `/eth/v2/beacon/blocks/{block_id}` for multiple milestone support.")
+            .tags(TAG_BEACON)
+            .pathParam(PARAMETER_BLOCK_ID)
+            .response(SC_OK, "Request successful", getResponseType(schemaDefinitionCache))
+            // .response(SC_BAD_REQUEST, "Bad request", HTTP_ERROR_RESPONSE_TYPE) // TODO do I need
+            // one for bad request?
```
Comment (courtneyeh) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlock.java]: Do we need a response for bad request?

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlock.java]: bad_request is auto-added, i think your test for 400 should pass without declaring it here

File: data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/migrated/schema/GetBlockResponse.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "data" : {
+      "title" : "SignedBeaconBlock",
+      "type" : "object",
+      "oneOf" : [ {
+        "$ref" : "#/components/schemas/SignedBeaconBlockPhase0"
+      }, {
+        "$ref" : "#/components/schemas/SignedBeaconBlockAltair"
+      }, {
+        "$ref" : "#/components/schemas/SignedBeaconBlockBellatrix"
+      } ]
```
Comment (rolfyone) [inline data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/migrated/schema/GetBlockResponse.json]: this getBlock only supports phase0...

Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/migrated/schema/GetBlockResponse.json]: its possibly that the name of the reutrned object needs to differ

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**